patches/server/0965-Make-banners-modifiable-with-both-BannerMeta-and-Blo.patch

```diff
@@ -0,0 +1,270 @@
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.add(pattern);
+     }
+ 
+     @Override
+     public Pattern getPattern(int i) {
++        if (true) return this.bannerBlock.getPattern(i); // Paper - Use underlying block state
+         return this.patterns.get(i);
+     }
+ 
+     @Override
+     public Pattern removePattern(int i) {
++        // Paper start - Use underlying block state
++        if (true) {
++            Pattern removedPattern = this.bannerBlock.removePattern(i);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return removedPattern;
++        }
++        // Paper end - Use underlying block state
+         return this.patterns.remove(i);
+     }
+ 
+     @Override
+     public void setPattern(int i, Pattern pattern) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setPattern(i, pattern);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.set(i, pattern);
+     }
+ 
+     @Override
+     public int numberOfPatterns() {
++        if (true) return this.bannerBlock.numberOfPatterns(); // Paper - Use underlying block state
+         return this.patterns.size();
+     }
+ 
+@@ -191,6 +261,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+     ImmutableMap.Builder<String, Object> serialize(ImmutableMap.Builder<String, Object> builder) {
+         super.serialize(builder);
+ 
++        if (true) return builder; // Paper - Let CraftMetaBlockState serialize for us
+         if (this.base != null) {
+             builder.put(BASE.BUKKIT, this.base.toString());
+         }
+@@ -206,6 +277,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+     int applyHash() {
+         final int original;
+         int hash = original = super.applyHash();
++        if (true) return original; // Paper - CraftMetaBlockState's hash is good enough
+         if (this.base != null) {
+             hash = 31 * hash + this.base.hashCode();
+         }
+@@ -217,6 +289,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+ 
+     @Override
+     public boolean equalsCommon(CraftMetaItem meta) {
++        if (true) return super.equalsCommon(meta); // Paper - Only check CraftMetaBlockState's data
+         if (!super.equalsCommon(meta)) {
+             return false;
+         }
+@@ -230,11 +303,13 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+ 
+     @Override
+     boolean notUncommon(CraftMetaItem meta) {
++        if (true) return super.notUncommon(meta) && (meta instanceof CraftMetaBanner || this.blockEntityTag == null);  // Paper - Only check CraftMetaBlockState's data
```

Comment:
YOu don't need the second part of this, `super.notUncommon(meta)` should already cover that in `CraftMetaBlockState`.